### PR TITLE
Allow recent omnibus-software ruby version convention.

### PR DESCRIPTION
**Pull request by ksubrama**
This allows master to build at chef/omnibus-software#556

### CODE REVIEW

**Comment by ksubrama** [comment]
@jaym @btm @schisamo @chef/engineering-services

File: omnibus/config/projects/push-jobs-client.rb [hunk 1]
```diff
@@ -38,15 +38,12 @@
...
-override :ruby,           version: "2.1.6"
-######
-# Ruby 2.1/2.2 has an error on Windows - HTTPS gem downloads aren't working
-# https://bugs.ruby-lang.org/issues/11033
-# Going to leave 2.1.5 for now since there is a workaround
-override :'ruby-windows', version: "2.1.5"
-override :'ruby-windows-devkit', version: "4.7.2-20130224-1151"
-#override :'ruby-windows', version: "2.0.0-p451"
-######
+if windows?
```
Comment (jaym) [inline omnibus/config/projects/push-jobs-client.rb]: is this change required?

**Comment by manderson26** [comment]
Looks reasonable to me. Right now CI doesn't check that the msi-installed windows push client actually executes, so it might be worth making sure of that before merging.

File: omnibus/config/projects/push-jobs-client.rb (hunk 1 above)
Comment (ksubrama) [inline omnibus/config/projects/push-jobs-client.rb]: This is the core change needed.  The default version for :ruby is now "ruby-windows" which tells it to go with what ruby-windows wants instead of compiling ruby from scratch.  So it's imperative to not override the version of ruby to "2.1.6" on the windows builds.
